treeherder/webapp/api/performance_data.py

```diff
@@ -15,7 +19,17 @@ class PerformanceDataViewSet(viewsets.ViewSet):
     This view serves performance charts data
     """
 
+    def _calculate_etag(view_instance, view_method,
+                       request, args, kwargs):
+        if not kwargs.get('project') or request.QUERY_PARAMS.get('interval'):
```

Comment:
Can you make this condition more explicit with some parenthesis?
Looking at the code below I think you want both project and interval to be there, right?